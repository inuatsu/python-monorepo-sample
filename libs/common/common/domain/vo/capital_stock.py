from typing import Any

from pydantic import RootModel


class CapitalStock(RootModel[int]):
    @classmethod
    def to_int(cls, value: Any) -> "CapitalStock":
        if isinstance(value, str):
            return CapitalStock(int(value.replace(",", "")))
        elif isinstance(value, int):
            return CapitalStock(value)
        else:
            raise TypeError("Type of argument should be str or int")
