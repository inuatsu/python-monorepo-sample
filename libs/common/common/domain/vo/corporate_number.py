from re import fullmatch
from typing import Annotated, Any

from pydantic import BeforeValidator


def validate_type_and_length(value: Any) -> str:
    if not isinstance(value, str):
        raise TypeError("Type of value should be str")
    if not value.isdigit():
        raise ValueError("Value should be digit")
    if len(value) < 12 or len(value) > 13:
        raise ValueError("Length of given value should be 12 or 13")
    return value


def validate_corporate_registration_number(value: str) -> str:
    corporate_registration_number = value[-12:]
    government_institutions = "00001[123]"
    municipalities = "0000[23]0"
    registration_office_codes = [
        "0100",
        "0101",
        "0104",
        "0105",
        "0106",
        "0107",
        "0108",
        "0109",
        "0110",
        "0111",
        "0112",
        "0113",
        "0114",
        "0115",
        "0116",
        "0117",
        "0118",
        "0123",
        "0124",
        "0127",
        "0128",
        "0131",
        "0132",
        "0133",
        "0134",
        "0200",
        "0210",
        "0300",
        "0400",
        "0500",
        "0600",
        "0700",
        "0800",
        "0801",
        "0804",
        "0900",
        "1000",
        "1100",
        "1200",
        "1201",
        "1209",
        "1220",
        "1300",
        "1400",
        "1500",
        "1600",
        "1700",
        "1800",
        "1803",
        "1900",
        "2000",
        "2100",
        "2200",
        "2300",
        "2400",
        "2500",
        "2600",
        "2700",
        "2800",
        "2900",
        "2908",
        "3000",
        "3100",
        "3200",
        "3202",
        "3300",
        "3400",
        "3500",
        "3600",
        "3700",
        "3701",
        "3702",
        "3703",
        "3704",
        "3705",
        "3706",
        "3708",
        "3800",
        "3900",
        "4000",
        "4005",
        "4006",
        "4027",
        "4100",
        "4200",
        "4300",
        "4400",
        "4500",
        "4600",
        "4601",
        "4603",
        "4604",
        "4625",
        "4700",
        "4800",
        "4900",
        "5000",
    ]
    corporate_type_codes = ["01", "02", "03", "04", "05"]
    corporations_with_establishment_registration = (
        rf"({'|'.join(registration_office_codes)})({'|'.join(corporate_type_codes)})"
    )
    corporations_without_establishment_registration = r"7\d{5}"
    match = fullmatch(
        rf"({government_institutions}|{municipalities}|{corporations_with_establishment_registration}|{corporations_without_establishment_registration})\d{{6}}",
        corporate_registration_number,
    )
    if match:
        return value
    else:
        raise ValueError("Given value is not valid corporate number")


def validate_check_digit(value: str) -> str:
    corporate_registration_number = value[-12:]
    even_digit_sum = (
        int(corporate_registration_number[0])
        + int(corporate_registration_number[2])
        + int(corporate_registration_number[4])
        + int(corporate_registration_number[6])
        + int(corporate_registration_number[8])
        + int(corporate_registration_number[10])
    )
    odd_digit_sum = (
        int(corporate_registration_number[1])
        + int(corporate_registration_number[3])
        + int(corporate_registration_number[5])
        + int(corporate_registration_number[7])
        + int(corporate_registration_number[9])
        + int(corporate_registration_number[11])
    )
    check_digit = 9 - (even_digit_sum * 2 + odd_digit_sum) % 9
    corporate_number = f"{check_digit}{corporate_registration_number}"
    if len(value) == 13:
        assert corporate_number == value, "Check digit is invalid"
    return corporate_number


CorporateNumber = Annotated[
    str,
    BeforeValidator(validate_check_digit),
    BeforeValidator(validate_corporate_registration_number),
    BeforeValidator(validate_type_and_length),
]
