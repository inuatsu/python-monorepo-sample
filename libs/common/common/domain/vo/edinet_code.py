from typing import Annotated

from pydantic import Field

EdinetCode = Annotated[str, Field(max_length=6, min_length=6, pattern=r"^E\d{5}$", frozen=True)]
