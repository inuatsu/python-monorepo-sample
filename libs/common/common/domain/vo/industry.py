from enum import Enum
from typing import Any

from pydantic import RootModel


class TSESector(Enum):
    FISHERY_AGRICULTURE_FORESTRY = "水産・農林業"
    FOODS = "食料品"
    MINING = "鉱業"
    OIL_COAL_PRODUCTS = "石油・石炭製品"
    CONSTRUCTION = "建設業"
    METAL_PRODUCTS = "金属製品"
    GLASS_CERAMICS_PRODUCTS = "ガラス・土石製品"
    TEXTILES_APPARELS = "繊維製品"
    PULP_PAPER = "パルプ・紙"
    CHEMICALS = "化学"
    PHARMACEUTICAL = "医薬品"
    RUBBER_PRODUCTS = "ゴム製品"
    TRANSPORTATION_EQUIPMENT = "輸送用機器"
    IRON_STEEL = "鉄鋼"
    NONFERROUS_METALS = "非鉄金属"
    MACHINERY = "機械"
    ELECTRIC_APPLIANCES = "電気機器"
    PRECISION_INSTRUMENTS = "精密機器"
    OTHER_PRODUCTS = "その他製品"
    INFORMATION_COMMUNICATION = "情報・通信業"
    SERVICES = "サービス業"
    ELECTRIC_POWER_GAS = "電気・ガス業"
    LAND_TRANSPORTATION = "陸運業"
    MARINE_TRANSPORTATION = "海運業"
    AIR_TRANSPORTATION = "空運業"
    WAREHOUSING_HARBOR_TRANSPORTATION = "倉庫・運輸関連"
    WHOLESALE_TRADE = "卸売業"
    RETAIL_TRADE = "小売業"
    BANKS = "銀行業"
    SECURITIES_COMMODITIES_FUTURES = "証券、商品先物取引業"
    INSURANCE = "保険業"
    OTHER_FINANCING_BUSINESS = "その他金融業"
    REAL_ESTATE = "不動産業"


class Industry(RootModel[TSESector]):
    @classmethod
    def from_str(cls, string: Any) -> "Industry | None":
        if not isinstance(string, str):
            raise TypeError("Type of argument should be str")
        for member in TSESector:
            if string == member.value:
                return Industry(member)
        return None
