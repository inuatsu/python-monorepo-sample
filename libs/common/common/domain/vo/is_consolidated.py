from enum import Enum
from typing import Any

from pydantic import RootModel


class Consolidated(Enum):
    CONSOLIDATED = True
    NON_CONSOLIDATED = False


class IsConsolidated(RootModel[Consolidated]):
    @classmethod
    def from_str(cls, string: Any) -> "IsConsolidated":
        if not isinstance(string, str):
            raise TypeError("Type of argument should be str")
        if string.lower() in ["有", "consolidated", "true"]:
            return IsConsolidated(Consolidated.CONSOLIDATED)
        elif string.lower() in ["無", "non_consolidated", "false"]:
            return IsConsolidated(Consolidated.NON_CONSOLIDATED)
        else:
            raise NotImplementedError
