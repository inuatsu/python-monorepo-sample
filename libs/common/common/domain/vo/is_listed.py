from enum import Enum
from typing import Any

from pydantic import RootModel


class Listed(Enum):
    LISTED = True
    UNLISTED = False


class IsListed(RootModel[Listed]):
    @classmethod
    def from_str(cls, string: Any) -> "IsListed":
        if not isinstance(string, str):
            raise TypeError("Type of argument should be str")
        if string.lower() in ["上場", "listed", "true"]:
            return IsListed(Listed.LISTED)
        elif string.lower() in ["非上場", "unlisted", "false"]:
            return IsListed(Listed.UNLISTED)
        else:
            raise NotImplementedError
