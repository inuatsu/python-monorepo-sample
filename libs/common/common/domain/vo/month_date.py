import re
from calendar import isleap, monthrange
from datetime import date
from re import Match

from pydantic import BaseModel, computed_field


class MonthDate(BaseModel):
    text: str

    def regex_search(self) -> Match | None:
        text = re.sub(r"\s", "", self.text)
        return re.search(r"(?P<month>\d{,2})[月\-/ー／](?P<date>末|\d{,2})日?", text)

    @computed_field
    @property
    def month(self) -> int | None:
        m = self.regex_search()
        if not m:
            return None
        if not m["month"].isdigit():
            return None
        return int(m["month"])

    @computed_field
    @property
    def date(self) -> int | None:
        m = self.regex_search()
        if not m:
            return None
        if m["date"] == "末":
            this_year = date.today().year
            if not m["month"].isdigit():
                return None
            month = int(m["month"])
            if month == 2:
                return monthrange(this_year, month)[1] - isleap(this_year)
            return monthrange(this_year, month)[1]
        if not m["date"].isdigit():
            return None
        return int(m["date"])
