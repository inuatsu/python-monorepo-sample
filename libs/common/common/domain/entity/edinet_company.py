from pydantic import BaseModel, Field, computed_field

from common.domain.vo.capital_stock import CapitalStock
from common.domain.vo.corporate_number import CorporateNumber
from common.domain.vo.edinet_code import EdinetCode
from common.domain.vo.industry import Industry
from common.domain.vo.is_consolidated import IsConsolidated
from common.domain.vo.is_listed import IsListed
from common.domain.vo.month_date import MonthDate
from common.domain.vo.ticker_symbol import TickerSymbol


class EdinetCompany(BaseModel):
    edinet_code: EdinetCode
    submitter_type: str
    is_listed: IsListed | None = Field(default=None)
    is_consolidated: IsConsolidated | None = Field(default=None)
    capital_stock: CapitalStock | None = Field(default=None)
    account_closing_month_date: MonthDate | None = Field(default=None, exclude=True)
    name: str
    english_name: str | None = Field(default=None)
    name_kana: str | None = Field(default=None)
    address: str | None = Field(default=None)
    industry: Industry | None = Field(default=None)
    ticker_symbol: TickerSymbol | None = Field(default=None)
    corporate_number: CorporateNumber | None = Field(default=None)

    @computed_field
    @property
    def account_closing_month(self) -> int | None:
        if self.account_closing_month_date:
            return self.account_closing_month_date.month
        return None

    @computed_field
    @property
    def account_closing_date(self) -> int | None:
        if self.account_closing_month_date:
            return self.account_closing_month_date.date
        return None
