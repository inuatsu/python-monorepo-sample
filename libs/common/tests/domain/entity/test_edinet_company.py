from typing import NamedTuple

from pydantic import ValidationError
from pytest import mark, raises

from common.domain.entity.edinet_company import EdinetCompany
from common.domain.vo.capital_stock import CapitalStock
from common.domain.vo.industry import Industry
from common.domain.vo.is_consolidated import IsConsolidated
from common.domain.vo.is_listed import IsListed
from common.domain.vo.month_date import MonthDate


class EdinetCompanyParam(NamedTuple):
    edinet_code: str
    submitter_type: str
    is_listed: IsListed | None
    is_consolidated: IsConsolidated | None
    capital_stock: CapitalStock | None
    account_closing_month_date: MonthDate | None
    name: str
    english_name: str | None
    name_kana: str | None
    address: str | None
    industry: Industry | None
    ticker_symbol: str | None
    corporate_number: str | None


class InvalidIsListedParam(NamedTuple):
    exception_class: type[BaseException]
    exception_message: str
    is_listed: str | list[str]


class InvalidIsConsolidatedParam(NamedTuple):
    exception_class: type[BaseException]
    exception_message: str
    is_consolidated: str | list[str]


class InvalidCorporateNumberParam(NamedTuple):
    exception_class: type[BaseException]
    exception_message: str
    corporate_number: str


class TestEdinetCompany:
    valid_edinet_company_params = [
        EdinetCompanyParam(
            edinet_code="E00001",
            submitter_type="内国法人・組合",
            is_listed=IsListed.from_str("上場"),
            is_consolidated=IsConsolidated.from_str("有"),
            capital_stock=CapitalStock.to_int(1000),
            account_closing_month_date=MonthDate(text="12月31日"),
            name="サンプル株式会社",
            english_name="Sample Co., Ltd.",
            name_kana="サンプルカブシキガイシャ",
            address="東京都千代田区千代田１−１",
            industry=Industry.from_str("水産・農林業"),
            ticker_symbol="11110",
            corporate_number="7010001000000",
        ),
        EdinetCompanyParam(
            edinet_code="E00009",
            submitter_type="内国法人・組合（有価証券報告書等の提出義務者以外）",
            is_listed=IsListed.from_str("非上場"),
            is_consolidated=IsConsolidated.from_str("有"),
            capital_stock=CapitalStock.to_int("1000"),
            account_closing_month_date=MonthDate(text="10 月末日"),
            name="株式会社テスト",
            english_name="Test Co., Ltd.",
            name_kana="カブシキガイシャテスト",
            address="東京都千代田区千代田１−１",
            industry=Industry.from_str("情報・通信業"),
            ticker_symbol=None,
            corporate_number="010001000001",
        ),
        EdinetCompanyParam(
            edinet_code="E10001",
            submitter_type="外国法人・組合",
            is_listed=IsListed.from_str("非上場"),
            is_consolidated=IsConsolidated.from_str("無"),
            capital_stock=CapitalStock.to_int("1,000"),
            account_closing_month_date=MonthDate(text="2月末日"),
            name="テスト株式会社",
            english_name="Test Co., Ltd.",
            name_kana="テストカブシキガイシャ",
            address="東京都千代田区千代田１−１",
            industry=Industry.from_str("外国法人・組合"),
            ticker_symbol=None,
            corporate_number="2700150000000",
        ),
        EdinetCompanyParam(
            edinet_code="E20001",
            submitter_type="個人（組合発行者を除く）",
            is_listed=None,
            is_consolidated=None,
            capital_stock=CapitalStock.to_int("0"),
            account_closing_month_date=None,
            name="田中　太郎",
            english_name=None,
            name_kana="タナカ　タロウ",
            address=None,
            industry=Industry.from_str("個人（組合発行者を除く）"),
            ticker_symbol=None,
            corporate_number=None,
        ),
    ]

    @mark.parametrize("param", valid_edinet_company_params)
    def test_valid(self, param: EdinetCompanyParam, snapshot):
        assert (
            EdinetCompany(
                edinet_code=param.edinet_code,
                submitter_type=param.submitter_type,
                is_listed=param.is_listed,
                is_consolidated=param.is_consolidated,
                capital_stock=param.capital_stock,
                account_closing_month_date=param.account_closing_month_date,
                name=param.name,
                english_name=param.english_name,
                name_kana=param.name_kana,
                address=param.address,
                industry=param.industry,
                ticker_symbol=param.ticker_symbol,
                corporate_number=param.corporate_number,
            )
            == snapshot
        )

    def test_invalid_edinet_code(self):
        with raises(ValidationError):
            EdinetCompany(edinet_code="A00001", submitter_type="内国法人・組合", name="テスト株式会社")

    invalid_is_listed_params = [
        InvalidIsListedParam(
            exception_class=TypeError, exception_message="Type of argument should be str", is_listed=["テスト"]
        ),
        InvalidIsListedParam(exception_class=NotImplementedError, exception_message="", is_listed="テスト"),
    ]

    @mark.parametrize("param", invalid_is_listed_params)
    def test_invalid_is_listed(self, param: InvalidIsListedParam):
        with raises(param.exception_class, match=param.exception_message):
            EdinetCompany(
                edinet_code="E00001",
                submitter_type="内国法人・組合",
                is_listed=IsListed.from_str(param.is_listed),
                name="テスト株式会社",
            )

    invalid_is_consolidated_params = [
        InvalidIsConsolidatedParam(
            exception_class=TypeError, exception_message="Type of argument should be str", is_consolidated=["テスト"]
        ),
        InvalidIsConsolidatedParam(exception_class=NotImplementedError, exception_message="", is_consolidated="テスト"),
    ]

    @mark.parametrize("param", invalid_is_consolidated_params)
    def test_invalid_is_consolidated(self, param: InvalidIsConsolidatedParam):
        with raises(param.exception_class, match=param.exception_message):
            EdinetCompany(
                edinet_code="E00001",
                submitter_type="内国法人・組合",
                is_consolidated=IsConsolidated.from_str(param.is_consolidated),
                name="テスト株式会社",
            )

    def test_invalid_capital_stock(self):
        with raises(TypeError, match="Type of argument should be str or int"):
            EdinetCompany(
                edinet_code="E00001",
                submitter_type="内国法人・組合",
                capital_stock=CapitalStock.to_int(150.5),
                name="テスト株式会社",
            )

    @mark.parametrize("month_date_text", [("八月三十一日"), ("テスト"), ("二月末日")])
    def test_invalid_account_closing_date(self, month_date_text: str, snapshot):
        assert (
            EdinetCompany(
                edinet_code="E00001",
                submitter_type="内国法人・組合",
                account_closing_month_date=MonthDate(text=month_date_text),
                name="テスト株式会社",
            )
            == snapshot
        )

    def test_invalid_industry(self):
        with raises(TypeError, match="Type of argument should be str"):
            EdinetCompany(
                edinet_code="E00001",
                submitter_type="内国法人・組合",
                name="テスト株式会社",
                industry=Industry.from_str(1),
            )

    def test_invalid_ticker_symbol(self):
        with raises(ValidationError):
            EdinetCompany(
                edinet_code="E00001", submitter_type="内国法人・組合", name="テスト株式会社", ticker_symbol="GOOGL"
            )

    invalid_corporate_number_params = [
        InvalidCorporateNumberParam(
            exception_class=TypeError, exception_message="Type of value should be str", corporate_number=1234567890123
        ),
        InvalidCorporateNumberParam(
            exception_class=ValueError, exception_message="Value should be digit", corporate_number="1234567890abc"
        ),
        InvalidCorporateNumberParam(
            exception_class=ValueError,
            exception_message="Length of given value should be 12 or 13",
            corporate_number="12345678901",
        ),
        InvalidCorporateNumberParam(
            exception_class=ValueError,
            exception_message="Length of given value should be 12 or 13",
            corporate_number="12345678901234",
        ),
        InvalidCorporateNumberParam(
            exception_class=ValueError,
            exception_message="Given value is not valid corporate number",
            corporate_number="1234567890123",
        ),
    ]

    @mark.parametrize("params", invalid_corporate_number_params)
    def test_invalid_corporate_number(self, params: InvalidCorporateNumberParam):
        with raises(params.exception_class, match=params.exception_message):
            EdinetCompany(
                edinet_code="E00001",
                submitter_type="内国法人・組合",
                name="テスト株式会社",
                corporate_number=params.corporate_number,
            )
