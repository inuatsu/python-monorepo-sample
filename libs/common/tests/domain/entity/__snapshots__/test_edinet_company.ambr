# serializer version: 1
# name: TestEdinetCompany.test_invalid_account_closing_date[テスト]
  EdinetCompany(edinet_code='E00001', submitter_type='内国法人・組合', is_listed=None, is_consolidated=None, capital_stock=None, account_closing_month_date=MonthDate(text='テスト', month=None, date=None), name='テスト株式会社', english_name=None, name_kana=None, address=None, industry=None, ticker_symbol=None, corporate_number=None, account_closing_month=None, account_closing_date=None)
# ---
# name: TestEdinetCompany.test_invalid_account_closing_date[二月末日]
  EdinetCompany(edinet_code='E00001', submitter_type='内国法人・組合', is_listed=None, is_consolidated=None, capital_stock=None, account_closing_month_date=MonthDate(text='二月末日', month=None, date=None), name='テスト株式会社', english_name=None, name_kana=None, address=None, industry=None, ticker_symbol=None, corporate_number=None, account_closing_month=None, account_closing_date=None)
# ---
# name: TestEdinetCompany.test_invalid_account_closing_date[八月三十一日]
  EdinetCompany(edinet_code='E00001', submitter_type='内国法人・組合', is_listed=None, is_consolidated=None, capital_stock=None, account_closing_month_date=MonthDate(text='八月三十一日', month=None, date=None), name='テスト株式会社', english_name=None, name_kana=None, address=None, industry=None, ticker_symbol=None, corporate_number=None, account_closing_month=None, account_closing_date=None)
# ---
# name: TestEdinetCompany.test_valid[param0]
  EdinetCompany(edinet_code='E00001', submitter_type='内国法人・組合', is_listed=IsListed(root=<Listed.LISTED: True>), is_consolidated=IsConsolidated(root=<Consolidated.CONSOLIDATED: True>), capital_stock=CapitalStock(root=1000), account_closing_month_date=MonthDate(text='12月31日', month=12, date=31), name='サンプル株式会社', english_name='Sample Co., Ltd.', name_kana='サンプルカブシキガイシャ', address='東京都千代田区千代田１−１', industry=Industry(root=<TSESector.FISHERY_AGRICULTURE_FORESTRY: '水産・農林業'>), ticker_symbol='11110', corporate_number='7010001000000', account_closing_month=12, account_closing_date=31)
# ---
# name: TestEdinetCompany.test_valid[param1]
  EdinetCompany(edinet_code='E00009', submitter_type='内国法人・組合（有価証券報告書等の提出義務者以外）', is_listed=IsListed(root=<Listed.UNLISTED: False>), is_consolidated=IsConsolidated(root=<Consolidated.CONSOLIDATED: True>), capital_stock=CapitalStock(root=1000), account_closing_month_date=MonthDate(text='10 月末日', month=10, date=31), name='株式会社テスト', english_name='Test Co., Ltd.', name_kana='カブシキガイシャテスト', address='東京都千代田区千代田１−１', industry=Industry(root=<TSESector.INFORMATION_COMMUNICATION: '情報・通信業'>), ticker_symbol=None, corporate_number='6010001000001', account_closing_month=10, account_closing_date=31)
# ---
# name: TestEdinetCompany.test_valid[param2]
  EdinetCompany(edinet_code='E10001', submitter_type='外国法人・組合', is_listed=IsListed(root=<Listed.UNLISTED: False>), is_consolidated=IsConsolidated(root=<Consolidated.NON_CONSOLIDATED: False>), capital_stock=CapitalStock(root=1000), account_closing_month_date=MonthDate(text='2月末日', month=2, date=28), name='テスト株式会社', english_name='Test Co., Ltd.', name_kana='テストカブシキガイシャ', address='東京都千代田区千代田１−１', industry=None, ticker_symbol=None, corporate_number='2700150000000', account_closing_month=2, account_closing_date=28)
# ---
# name: TestEdinetCompany.test_valid[param3]
  EdinetCompany(edinet_code='E20001', submitter_type='個人（組合発行者を除く）', is_listed=None, is_consolidated=None, capital_stock=CapitalStock(root=0), account_closing_month_date=None, name='田中\u3000太郎', english_name=None, name_kana='タナカ\u3000タロウ', address=None, industry=None, ticker_symbol=None, corporate_number=None, account_closing_month=None, account_closing_date=None)
# ---
